sdks/python/apache_beam/io/fileio.py

```diff
@@ -728,10 +729,14 @@ def _fetch_to_internal_buffer(self, num_bytes):
             pass  # All is as expected!
         else:
           self._data += self._decompressor.flush()
+        # Record that we have hit the end of file, so we won't unnecessarily
+        # repeat the completeness verification step above.
+        self._read_eof = True
         return
 
   def _read_from_internal_buffer(self, num_bytes):
     """Read up to num_bytes from the internal buffer."""
+    # TODO: this can be optimized to avoid a string copy operation.
```

Comment:
driveby that many of these TODOs make good JIRA issues, especially to empower new potential contributors to the SDK.